### PR TITLE
[AMDGPU] Cleanup bitcast spam in atomic optimizer

### PR DESCRIPTION
**Pull request by vikramRH**

### CODE REVIEW

**Comment by vikramRH** [comment]
* **#96933** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/96933?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vikramRH and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by arsenm — APPROVED** [review]

**Review by pravinjagtap — APPROVED** [review]
LGTM.
Code looks much cleaner now.

File: llvm/lib/Target/AMDGPU/AMDGPUAtomicOptimizer.cpp [hunk 1]
```diff
@@ -922,26 +887,8 @@ void AMDGPUAtomicOptimizerImpl::optimizeAtomic(Instruction &I,
...
-      Value *const Insert =
-          B.CreateInsertElement(PartialInsert, ReadFirstLaneHi, B.getInt32(1));
-      BroadcastI = B.CreateBitCast(Insert, Ty);
-    } else if (TyBitWidth == 32) {
-      Value *CastedPhi = B.CreateBitCast(PHI, IntNTy);
-      BroadcastI =
-          B.CreateIntrinsic(IntNTy, Intrinsic::amdgcn_readfirstlane, CastedPhi);
-      BroadcastI = B.CreateBitCast(BroadcastI, Ty);
-
+    if (TyBitWidth == 32 || TyBitWidth == 64) {
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/AMDGPUAtomicOptimizer.cpp:890]: Just remove the "if" and call the CreateIntrinsic unconditionally.

Comment (vikramRH) [reply llvm/lib/Target/AMDGPU/AMDGPUAtomicOptimizer.cpp]: done

**Review by arsenm — APPROVED** [review]

**Pull request merged**